### DISCUSSION TITLE
Error installing version 9.0.0

### DISCUSSION THREAD

**Discussion by daniel89fg** [spatie/laravel-medialibrary#2400] [Q&A]
Hello.

I just did a clean install of Xampp. I have enabled the GD library. When trying to make the composer require "spatie/laravel-medialibrary:^9.0.0" I have the following error and cannot continue.

Installation summary: xampp 8.0.3 + php 8.0.3 + 10.4.18-MariaDB + Apache / 2.4.46

What is the problem, please help.

./composer.json has been updated
Running composer update spatie/laravel-medialibrary
Loading composer repositories with package information
Updating dependencies
Lock file operations: 7 installs, 0 updates, 0 removals
  - Locking league/glide (1.7.0)
  - Locking maennchen/zipstream-php (2.1.0)
  - Locking myclabs/php-enum (1.8.0)
  - Locking spatie/image (1.10.5)
  - Locking spatie/image-optimizer (1.3.2)
  - Locking spatie/laravel-medialibrary (9.0.0)        
  - Locking spatie/temporary-directory (1.3.0)
Writing lock file
Installing dependencies from lock file (including require-dev)
Package operations: 7 installs, 0 updates, 0 removals
  - Installing league/glide (1.7.0): Extracting archive
  - Installing myclabs/php-enum (1.8.0): Extracting archive
  - Installing spatie/image-optimizer (1.3.2): Extracting archive
  - Installing spatie/temporary-directory (1.3.0): Extracting archive
  - Installing spatie/image (1.10.5): Extracting archive
  - Installing maennchen/zipstream-php (2.1.0): Extracting archive
  - Installing spatie/laravel-medialibrary (9.0.0): Extracting archive
3 package suggestions were added by new dependencies, use `composer suggest` to see details.
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
> @php artisan package:discover --ansi

   Error

  Class name must be a valid object or a string

  at C:\xampp\htdocs\laravelbase\vendor\spatie\laravel-medialibrary\src\MediaLibraryServiceProvider.php:23
     19▕         $this->registerPublishables();
     20▕
     21▕         $mediaClass = config('media-library.media_model');
     22▕
  ➜  23▕         $mediaClass::observe(new MediaObserver());
     24▕
     25▕         $this->loadViewsFrom(__DIR__.'/../resources/views', 'media-library');
     26▕     }
     27▕

  1   C:\xampp\htdocs\laravelbase\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php:36  
      Spatie\MediaLibrary\MediaLibraryServiceProvider::boot()

  2   C:\xampp\htdocs\laravelbase\vendor\laravel\framework\src\Illuminate\Container\Util.php:40
      Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
Script @php artisan package:discover --ansi handling the post-autoload-dump event returned with error code 1

Installation failed, reverting ./composer.json and ./composer.lock to their original content.

**Comment by BrandonSurowiec** [comment]
So the reason this happens (I've run into this as well) is because you have your config cached when you pulled in this package so `$mediaClass = config('media-library.media_model');` is null.

I've run into this quite often recently as I forget we're caching the config on our deploy script. Because the config is already cached when Laravel runs the following line in the Spatie Service provider, the media library config doesn't get merged in even though it needs to be.

```
$this->mergeConfigFrom(__DIR__.'/../config/media-library.php', 'media-library');
```

You'll need to clear your config with `php artisan config:clear`. Now in the current state, this command will fail as Laravel cannot boot up. So you'll need to go into your `/bootstrap/cache` folder of your application and delete the `config.php` file to clear the cache manually.

**Reply by daniel89fg** [reply]
IT WORKED ! Thank you. I will get used to clearing the cache often.

**Reply by BrandonSurowiec** [reply]
For local development it is highly recommended not to run `config:cache`. Happy coding :)
